mllib/src/test/scala/org/apache/spark/ml/util/TempDirectory.scala

```diff
@@ -35,11 +35,11 @@ trait TempDirectory extends BeforeAndAfterAll { self: Suite =>
 
   override def beforeAll(): Unit = {
     super.beforeAll()
-    _tempDir = Utils.createTempDir(this.getClass.getName)
+    _tempDir = Utils.createTempDir(namePrefix = this.getClass.getName)
   }
 
   override def afterAll(): Unit = {
-    Utils.deleteRecursively(_tempDir)
+    Utils.deleteRecursively(_tempDir.getParentFile)
```

Comment:
Do we still need to remove parent? Now the root is `${java.io.tmp}`, which is created by default and the temp directory is `o.a.s.ml.feature.ABC-UUID`. So we should only remove `_tempDir` instead of its parent.